### DISCUSSION TITLE
SVG `fill-transparent` always takes precedence?

### DISCUSSION THREAD

**Discussion by r-i-c-h** [tailwindlabs/tailwindcss#11398] [Help]
In a TS-React app I'm trying to conditionally change the fill of an SVG Icon. 

I want the "default" state to be `fill-transparent` but want to have the ability to use a className prop in order to change the fill. 

I see in earlier versions of TW, one would apply `fill-current text-blue-500`, and now we have the `fill-blue-500` - No matter how I set things up, `fill-transparent` *always* dominates. See https://play.tailwindcss.com/JHtrQuIPqc

At the end of the day I know this is about CSS-Specificity, but I'm wondering if someone could suggest a "good" way to conditionally change the SVG icon's fill color? 

Example of what I'm doing...
```
export function SomeIcon(props: SVGAttributes<SVGElement>) {
  return (
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32" aria-hidden="true" strokeLinecap="round" strokeLinejoin="round" {...props}
      className={`fill-transparent stroke-2 stroke-current h-6 w-6 ${props.className ?? ''}`}
    ><path "yada-yada..."/>
    </svg>
  )
}

/* Implementation: */
<ParentThing>
  <SomeIcon className="fill-red-500" />
<ParentThing/>
```

**Comment by wongjn** [comment]
Ensure only one utility class exists on an element that applies the same CSS property with the same variants, hence try:

```js
className={`stroke-2 stroke-current h-6 w-6 ${props.className ?? 'fill-transparent'}`}
```
Or if there might be other classes in `className`:
```tsx
export function SomeIcon(props: SVGAttributes<SVGElement>) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 32 32"
      aria-hidden="true"
      strokeLinecap="round"
      strokeLinejoin="round"
      {...props}
      className={`${props.fill ?? 'fill-transparent'} stroke-2 stroke-current h-6 w-6 ${props.className ?? ''}`}
    >
      <path "yada-yada..."/>
    </svg>
  )
}
```
```html
<ParentThing>
  <SomeIcon fill="fill-red-500" />
<ParentThing/>
```
